src/main/java/io/javalin/Javalin.java

```diff
@@ -123,21 +123,13 @@ public Javalin start() {
             eventManager.fireEvent(JavalinEvent.SERVER_STARTING);
             try {
                 log.info("Starting Javalin ...");
-                JavalinServlet javalinServlet = new JavalinServlet(
-                    this,
-                    pathMatcher,
-                    exceptionMapper,
-                    errorMapper,
-                    debugLogging,
-                    requestLogger,
-                    dynamicGzipEnabled,
-                    autogeneratedEtagsEnabled,
-                    defaultContentType,
-                    maxRequestCacheBodySize,
-                    prefer405over404,
-                    singlePageHandler,
-                    new JettyResourceHandler(staticFileConfig, jettyServer, ignoreTrailingSlashes)
-                );
+                if (jettyServer == null) {
+                    jettyServer = JettyServerUtil.defaultServer();
+                }
+                JavalinServlet javalinServlet = createServlet(new JettyResourceHandler(staticFileConfig, jettyServer, ignoreTrailingSlashes));
```

Comment:
Could you move this under the other null-check?

Reply:
Passing `jettyServer` and `jettySessionHandler` as constructor parameters seem to work pretty nicely, so I was able to remove the now-useless null check.